code/parse/sexp_container.cpp

```diff
@@ -1311,6 +1320,68 @@ void sexp_clear_container(int node)
 	}
 }
 
+void sexp_copy_container(int node)
+{
+	Assertion(node != -1, "Copy-container wasn't given a container to copy from. Please report!");
+
+	const char *src_container_name = CTEXT(node);
+	const auto *p_src_container = get_sexp_container(src_container_name);
+
+	if (!p_src_container) {
+		report_nonexistent_container("Copy-container", src_container_name);
+		return;
+	}
+	const auto &src_container = *p_src_container;
+
+	node = CDR(node);
+	Assertion(node != -1, "Copy-container wasn't given a container to copy to. Please report!");
+
+	const char *dest_container_name = CTEXT(node);
+	auto *p_dest_container = get_sexp_container(src_container_name);
+
+	if (!p_dest_container) {
+		report_nonexistent_container("Copy-container", dest_container_name);
+		return;
+	}
+	auto &dest_container = *p_dest_container;
+
+	if (src_container.get_non_persistent_type() != dest_container.get_non_persistent_type()) {
```

Comment:
Seems pretty straight forward.  The only thing is the name of this function `get_non_persistent_type`.  That just means what type of container type it is?

Reply:
Yeah, get a container's type (function return type is `enum class ContainerType`) with persistence flags zeroed out.

See the SEXP help text and also the use case described in #fred for why this is done.

Reply:
Admittedly `get_non_persistent_Type()` wasn't the best function name. 🤷‍♂️

Reply:
Go figure